### ISSUE TITLE
Brace constructing a toml::value inserts a new array element in GCC 10.2.1

### ISSUE THREAD

**Issue by andrewkcorcoran** [ToruNiina/toml11#165]
toml11v3.7.0

The `toml2` variable in the below example is copy constructed incorrectly and has an empty array inserted on GCC but is correctly constructed on clang. The error looks to be to do with the `toml::value` `std::intializer_list` constructor `std::enable_if` being too broad (toml/value.hpp:840).

```cpp
  #include <iostream>
  #include <sstream>

  #include "toml.hpp"

  int main()
  {
      auto stream = std::istringstream{"a = 1"};
      auto toml1 = toml::parse(stream);
      auto toml2 = toml::value{toml1};
      auto toml3 = toml::value(toml1);
      std::cout << toml1 << toml2 << toml3;
      return EXIT_SUCCESS;
  }
```

```bash
$ g++ --version
g++ (GCC) 10.2.1 20200804 (Red Hat 10.2.1-2)
./mva_gcc
a = 1
[[]]
a = 1
a = 1

$ clang++ --version
clang version 10.0.1 (Red Hat 10.0.1-4.el7)
./mva_clang
a = 1
a = 1
a = 1
```

**Comment by jackwil1** [comment]
I am also having this issue

**Comment by vadz** [comment]
This is really not a nice problem to run into when initializing a class member with the result of `parse()`, e.g.

```cpp
struct config_file {
  explicit config_file(const char* filename) : root_{toml::parse(filename)} {}

  const toml::value root_;
};
```

results in `root_` being an array, completely unexpectedly.

**Comment by ToruNiina** [comment]
This issue is difficult to resolve due to the syntax of C++, and it seems that other libraries have not been able to fully resolve it either. This includes the very well-known library, nlohmann/json.

https://json.nlohmann.me/home/faq/#known-bugs

I just released v4.0.0. After considering several approaches, I decided to stop supporting `std::initializer_list` because of this issue. Aside from unintentionally creating arrays, there is also the problem of not being able to distinguish between a table and nested arrays (I mean, `toml::value{{"key1", "value1"}, {"key2", "value2"}}`). Considering the negative impact, we concluded that it is somewhat better to require the explicit use of `toml::array` and `toml::table` each time if those are needed.

For more detailed explanations, please refer to [the documentation](https://toruniina.github.io/toml11/docs/changelog/#removed-stdinitializer_list-support-for-tomlbasic_value).
